include/Dialect/LWE/IR/LWEOps.td

```diff
@@ -77,17 +80,33 @@ def LWE_RLWEEncodeOp : LWE_Op<"rlwe_encode", [Pure]> {
   }];
 
   let arguments = (ins
-    SignlessIntegerLike:$plaintext,
+    SignlessIntegerLike:$input,
     AnyRLWEEncodingAttr:$encoding,
     Ring_Attr:$ring
   );
 
   let results = (outs RLWEPlaintext:$output);
-  let assemblyFormat = "$plaintext attr-dict `:` qualified(type($plaintext)) `to` qualified(type($output))";
+  let assemblyFormat = "$input attr-dict `:` qualified(type($input)) `->` qualified(type($output))";
 
   // Verify that the encoding and ring parameter matches the output plaintext attribute.
   let hasVerifier = 1;
 }
 
+def LWE_RLWEDecodeOp : LWE_Op<"rlwe_decode", [Pure]> {
+  let summary = "Decode an RLWE plaintext to an underlying type";
+
+  let arguments = (ins
+    RLWEPlaintext:$input,
+    AnyRLWEEncodingAttr:$encoding,
+    Ring_Attr:$ring
+  );
+
+  let results = (outs SignlessIntegerLike:$output);
+  let assemblyFormat = "$input attr-dict `:` qualified(type($input)) `->` qualified(type($output))";
+
+  // FIXME: Verify that the encoding and ring parameter matches the input
+  // plaintext attribute.
```

Comment:
will this be fixed in an upcoming PR?

Reply:
Ah, this was from my splitting the bigger PR. Let me fix it in this one.

Reply:
Fixed.